public/app/plugins/datasource/grafana-testdata-datasource/components/StreamingClientEditor.tsx

```diff
@@ -29,29 +30,41 @@ export const StreamingClientEditor = ({ onChange, query }: EditorProps) => {
     onChange({ target: { name, value: Number(value) } });
   };
 
+  const fields =
+    query?.stream?.type === 'signal'
```

Comment:
Readability of this logic might increase if it's broken out into a couple functions?

Reply:
I'd agree here, doing a guard to check that query and stream exist would make this a lot more readable.

Reply:
Changed it a bit but the problem here is typing. Typing the dynamic UI creation based on the query type and associated fields is a bit harder without adding some explicit casts and need lots of additional typing which I don't want to go into right now.

For example even changing this into `if else` would mean TS complains about assigning different types to `fields` variable, because the `streamingClientFields` is `as const` type which is needed for the field mapping to `query.stream.[field]` later on to work.